week1/index.js

```diff
@@ -1 +1,57 @@
-console.log("Hello world")
\ No newline at end of file
+import HTTP from 'http'
+import dflt from './resFun/dflt.js'
+import addFn from './resFun/addFn.js'
+import removeFn from './resFun/removeFn.js'
+import resetFn from './resFun/resetFn.js'
+// import Path from 'path'
+
+var state = 10
+let addOne = () => ++state
+let reState = () => state = 10
+let subOne = () => --state
+
+const server = HTTP.createServer((request, response)=>{
+    console.log(request.method, request.url)
+    switch (request.url){
+        case '/':
+        dflt(response)
+        break
+        case '/state':
+        response.write(state.toString())
+        break
+        case '/add':
+        addOne()
+        addFn(response)
+        response.write(state.toString())
+
+        break
+        case '/remove':
+        subOne()
+        removeFn(response)
+        response.write(state.toString())        
+
+        break
+        case '/reset':
+        reState()
+        resetFn(response)
+        response.write(state.toString())        
+        break
+        
+        default:
+        response.statusCode = 404
+        response.setHeader('Contetn-Type', 'text/file')
+        response.write('404 error -- the file is not found- Fo')
```

Comment:
The user message could be a bit more helpful here. Users of this api, they are not actually requesting for a file. They are performing an action. And you could still return as `text/html`!